### PR TITLE
Implement Github Actions to require passing Jest tests before PR merge is allowed

### PR DESCRIPTION
**Pull request by joncarlson**
As the title states, adds Github actions to the project.

This will do a fresh `npm install` and run Jest tests against Node 16. Any failures will block the PR from being merged until Jest tests are passing again.

See the bottom of this PR page for a "Successful checks" area. That is the Github actions output. It should say "All checks have passed".

### CODE REVIEW

File: packages/app/email-notifications/__tests__/__snapshots__/email-notifications.test.ts.snap [hunk 1]
```diff
@@ -8,7 +8,7 @@ An action is required to transition the document to one of the [Draft, Approved]
   "cc": Array [
     "\\"Eggs User\\" <eggs@mock.nasa.gov>",
   ],
-  "createdOn": "2022-01-01T05:00:00.000Z",
+  "createdOn": "2022-01-01T00:00:00.000Z",
```
Comment (joncarlson) [inline packages/app/email-notifications/__tests__/__snapshots__/email-notifications.test.ts.snap:11]: Switching timezone to GMT

File: .github/workflows/continuous-integration.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    branches-ignore:
+      - main
+
+jobs:
+  app-jest:
+    runs-on: ubuntu-latest
+    name: app-node-${{ matrix.node }}
+    strategy:
+      matrix:
+        node: [16] # One or more Node versions to test against
```
Comment (joncarlson) [inline .github/workflows/continuous-integration.yaml:16]: Only testing against Node 16 but this allows us to test against multiple versions if we wanted to

File: packages/app/jest.config.js [hunk 3]
```diff
@@ -2,6 +2,8 @@
 const tsPreset = require('ts-jest/jest-preset')
 const mongoPreset = require('@shelf/jest-mongodb/jest-preset')
 
+process.env.TZ = 'GMT' // ensure all tests use GMT time, both locally and on CI servers
```
Comment (benforshey) [inline packages/app/jest.config.js:5]: I tried so hard to make this work in `jest.setup.ts`, with all the other environment variables. 😆  Setting Jest to use a `globaConfig` file also didn't work (broke something with `@shelf/jest-mongodb`, I think). So all that is to say, I see why this is here.

File: .github/workflows/continuous-integration.yaml (hunk 2 above)
Comment (benforshey) [reply .github/workflows/continuous-integration.yaml:16]: Nice—very excited to see CI features being added in!

**Pull request merged**